swarm.go

```diff
@@ -235,6 +262,26 @@ func (s *Swarm) addConn(tc transport.CapableConn, dir network.Direction) (*Conn,
 	s.notifyAll(func(f network.Notifiee) {
 		f.Connected(s, c)
 	})
+
+	// ---------------------------------------------------------
+	// Emit the Connectedness=Connected event on the bus if we just went
+	// from having no connections with the peer to having this connection with the peer.
+	indexForLk := len(p) - 1
+	stripe := &s.stripedPeerConnectivity[p[indexForLk]]
+	stripe.Lock()
+	s.conns.RLock()
+	// TODO Probably being over-cautious here, need to clean up
+	cs := s.conns.m[p]
+	// we went from having no connection with the peer to having this connection with the peer
+	// if this is the ONLY open connection we have with the peer.
+	isConnectednessFlip := (len(cs) == 1 && cs[0] == c && !cs[0].conn.IsClosed())
```

Comment:
Same as below, we can't re-take the lock when we check this. We need to check if we're going from 0->1 and atomically take the notification lock before dropping the lock on the connection map.